src/main/java/ch/njol/skript/effects/EffHealth.java

```diff
@@ -31,105 +31,105 @@
-		double damage = 0;
-		if (this.damage != null) {
-			Number number = this.damage.getSingle(e);
-			if (number == null)
+	protected void execute(Event event) {
+		double amount = 0;
+		if (this.amount != null) {
+			Number amountPostCheck = this.amount.getSingle(event);
+			if (amountPostCheck == null)
 				return;
-			damage = number.doubleValue();
+			amount = amountPostCheck.doubleValue();
 		}
-		Object[] array = damageables.getArray(e);
-		Object[] newArray = new Object[array.length];
 
-		boolean requiresChange = false;
-		for (int i = 0; i < array.length; i++) {
-			Object value = array[i];
-			if (value instanceof ItemType) {
-				ItemType itemType = (ItemType) value;
-				ItemStack itemStack = itemType.getRandom();
+		for (Object obj : this.damageables.getArray(event)) {
+			if (obj instanceof ItemType) {
+				ItemType itemType = (ItemType) obj;
 
-				if (this.damage == null) {
+				if (this.amount == null) {
+					ItemUtils.setDamage(itemType, 0);
+				} else {
+					ItemUtils.setDamage(itemType, (int) Math2.fit(0, (ItemUtils.getDamage(itemType) + (isHealing ? -amount : amount)), itemType.getMaterial().getMaxDurability()));
+				}
+
+			} else if (obj instanceof Slot) {
+				Slot slot = (Slot) obj;
+				ItemStack itemStack = slot.getItem();
+
+				if (itemStack == null)
+					continue;
+
+				if (this.amount == null) {
 					ItemUtils.setDamage(itemStack, 0);
 				} else {
-					ItemUtils.setDamage(itemStack, (int) Math2.fit(0, ItemUtils.getDamage(itemStack) + (heal ? -damage : damage), itemStack.getType().getMaxDurability()));
+					int damageAmt = (int) Math2.fit(0, (isHealing ? -amount : amount), itemStack.getType().getMaxDurability());
+					ItemUtils.setDamage(itemStack, damageAmt);
 				}
 
-				newArray[i] = new ItemType(itemStack);
-				requiresChange = true;
-			} else {
-				LivingEntity livingEntity = (LivingEntity) value;
-				if (!heal) {
-					HealthUtils.damage(livingEntity, damage);
-				} else if (this.damage == null) {
-					HealthUtils.setHealth(livingEntity, HealthUtils.getMaxHealth(livingEntity));
+				slot.setItem(itemStack);
+
+			} else if (obj instanceof Damageable) {
+				Damageable damageable = (Damageable) obj;
+
+				if (this.amount == null) {
+					HealthUtils.heal(damageable, HealthUtils.getMaxHealth(damageable));
+				} else if (isHealing) {
+					HealthUtils.heal(damageable, amount);
 				} else {
-					HealthUtils.heal(livingEntity, damage);
+					HealthUtils.damage(damageable, amount);
 				}
 
-				newArray[i] = livingEntity;
 			}
 		}
-
-		if (requiresChange)
-			damageables.change(e, newArray, ChangeMode.SET);
 	}
 
 	@Override
-	public String toString(@Nullable Event e, boolean debug) {
-		return (heal ? "heal" : "damage") + " " + damageables.toString(e, debug) + (damage != null ? " by " + damage.toString(e, debug) : "");
+	public String toString(@Nullable Event event, boolean debug) {
+		return (isHealing ? "heal/repair" : "damage" + " " + damageables.toString(event, debug)
```

Comment:
Why is this saying `heal/repair` in the stringified syntax? It should choose the correct one to print based on its circumstances.

Reply:
there is no circumstances I can pick from... the value is `isHealing` for both `repair` and `heal` just like it was before, the usage of both being added is to clarify when someone does `repair player's tool by 5` not to show up as `heal player's tool by 5` in stack traces,

Reply:
They have different patterns so you could keep track of the pattern from init if you had to.